src/chfn.c

```diff
@@ -648,11 +648,20 @@ int main (int argc, char **argv)
 	 * name, or the name getlogin() returns.
 	 */
 	if (optind < argc) {
-		user = argv[optind];
+		if (is_valid_user_name (argv[optind])) {
+			user = xstrdup (argv[optind]);
+		}
+		else {
+			fprintf (stderr, _("%s: user '%s' does not exist\n"), Prog,
+			         user);
```

Comment:
This is undefined behavior.

ISO C says "`the argument shall be a pointer to the initial element of an array of character type`", and `NULL` doesn't comply with that.

Reply:
This is fixed